src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/pricing/importer/PricesSpreadsheet.kt

```diff
@@ -38,22 +36,18 @@ class PricesSpreadsheet(private val spreadsheet: Workbook,
 
     private fun getPrice(supplier: Supplier, row: Row): Price {
         val journeyId = Result.runCatching { row.getCell(JOURNEY_ID).numericCellValue }.getOrElse { throw RuntimeException("Error retrieving journey id for supplier '$supplier'", it) }
-        val fromLocationName = row.getFormattedStringCell(FROM_LOCATION)
-                ?: throw RuntimeException("From location name cannot be blank")
-        val toLocationName = row.getFormattedStringCell(TO_LOCATION)
-                ?: throw RuntimeException("To location name cannot be blank")
+
+        val fromLocationName = row.getFormattedStringCell(FROM_LOCATION) ?: throw RuntimeException("From location name cannot be blank")
+
+        val toLocationName = row.getFormattedStringCell(TO_LOCATION) ?: throw RuntimeException("To location name cannot be blank")
+
         val price = Result.runCatching { (row.getCell(PRICE).numericCellValue * 100).toInt() }
                 .onSuccess { if (it == 0) throw RuntimeException("Price must be greater than zero") }
                 .getOrElse { throw RuntimeException("Error retrieving price for supplier '$supplier'", it) }
 
-        val fromLocation = locationRepo.findBySiteName(fromLocationName)
-                ?: throw RuntimeException("From location '$fromLocationName' for supplier '$supplier' not found")
-        val toLocation = locationRepo.findBySiteName(toLocationName)
-                ?: throw RuntimeException("To location '$toLocationName' for supplier '$supplier' not found")
+        val fromLocation = locations[fromLocationName] ?: throw RuntimeException("From location '$fromLocationName' for supplier '$supplier' not found")
```

Comment:
Replaced repository read with an in-memory read.  There was only a slight gain here but it feels like really we should be reading like this here anyway.